app/src/main/java/com/lasockiquenon/coopertest/TestActivity.kt

```diff
@@ -48,23 +46,45 @@ class TestActivity : AppCompatActivity(), OnMapReadyCallback {
 
     private lateinit var locationService: LocationService
     private var isServiceBound = false
+    private var isTestRunning = false
 
+    private var printResultNumber : Int =-1
+    private var routePointsLatLng : List<LatLng> = emptyList()
+    var cu : CameraUpdate? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_test)
+        printResultNumber= getIntent().getIntExtra("Results", -1)
 
         val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
         mapFragment.getMapAsync(this)
 
-        mFusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
-        notifier = AudioNotifier(this)
-        testTimer = getTestTimer()
-        startTimer = getStartTimer()
-
-        val intent = Intent(this, LocationService::class.java)
-        bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE)
-        startService(intent)
+        if (printResultNumber!=-1){
```

Comment:
Could you turn that into a class variable that tells us why we do the code? Something like onlyShowingResultsNoTest or whatever better name. And then all these things connected to showing results as separate functions below. TestActivity is getting longer and longer and it would be nice if the test logit would still be clear